mp/src/game/shared/neo/weapons/weapon_neobasecombatweapon.cpp

```diff
@@ -1093,22 +1100,29 @@ int CNEOBaseCombatWeapon::DrawModel(int flags)
 		IMaterial* pass = materials->FindMaterial("dev/motion_third", TEXTURE_GROUP_MODEL);
 		modelrender->ForcedMaterialOverride(pass);
 		ret |= BaseClass::DrawModel(flags);
+		modelrender->ForcedMaterialOverride(null);
 	}
 
 	else if (inThermalVision && !pOwner->IsCloaked())
 	{
 		IMaterial* pass = materials->FindMaterial("dev/thermal_third", TEXTURE_GROUP_MODEL);
 		modelrender->ForcedMaterialOverride(pass);
 		ret |= BaseClass::DrawModel(flags);
+		modelrender->ForcedMaterialOverride(null);
 	}
 
-	modelrender->ForcedMaterialOverride(null);
 	return ret;
 }
 
 int CNEOBaseCombatWeapon::InternalDrawModel(int flags)
 {
 	int ret = 0;
```

Comment:
Nitpicking, but could we move this below line 1125 for clarity, since we do a return without using its retval for the added flags check. It's pretty clear currently but if code gets added later it might become nonobvious that it's bypassed by the early BaseClass return otherwise.